api/v1alpha4/awsmachine_webhook.go

```diff
@@ -149,6 +149,10 @@ func (r *AWSMachine) validateRootVolume() field.ErrorList {
 		allErrs = append(allErrs, field.Required(field.NewPath("spec.rootVolumeOptions.iops"), "iops required if type is 'io1' or 'io2'"))
 	}
 
+	if r.Spec.RootVolume.Throughput < 0 {
```

Comment:
I think we need a similar check in the webhooks for:
-  `AWSMachinePool` as it uses volumes via `AWSLaunchTemplate`.
-  `AWSMachineTemplate`

Reply:
Added validation for the root volume for `AWSMachinePool` and validation for root volume + non-root volumes for `AWSMachineTemplate`.